modules/opensearch/includes/opensearch.search.inc

```diff
@@ -98,5 +99,206 @@ function opensearch_search_search(TingSearchRequest $ting_query) {
 
   // Perform the Open Search search and wrap the result.
   return new OpenSearchTingSearchResult(ting_do_search($query, 1, 10, array('facets' => $ting_query->getFacets(), 'all Objects' => FALSE)));
+}
+
+/**
+ * Allows the provider to programtically filter relations
+ *
+ * @return array
+ *   Filtered list.
+ */
+function opensearch_search_filter_relations($relations) {
+  $relation_types = opensearch_search_inline_relation_types();
+
+  return array_filter($relations,
+    function($relation) use ($relation_types) {
+
+      if (!isset($relation_types[$relation->type])) {
+        return FALSE;
+      }
+      // The isPartOfManifestion relation has all relation to other libraries
+      // copies of the relation, so we filter the ones that do not belong to
+      // current library.
+      if ($relation->type == 'dbcbib:isPartOfManifestation' && is_object($relation->object) && $relation->object instanceof TingEntity && $relation->object->reply->isLocal()) {
+        return FALSE;
+      }
+      return TRUE;
+    }
+  );
+}
+
+/**
+ * Produce a list of relations we can handle.
+ *
+ * @return string[]
+ *   Array of supported type-titles keyed by type name.
+ */
+function opensearch_search_supported_relations() {
+  return [
+      'dbcaddi:hasCreatorDescription' => t('Author portrait'),  
+      'dbcaddi:hasReview' => t('Review'),
+      'dbcaddi:hasSubjectDescription' => t('Subject description'),
+      'dbcaddi:hasDescriptionFromPublisher' => t('Description from publisher'),
+      'dbcaddi:hasCreatorHomePage' => t('Creator homepage'),
+      'dbcbib:isPartOfManifestation' => t('Is part of manifestation'),
+  ];
+}
+
+/**
+ * Some relations are requested NOT to load a new window.
+ *
+ * @param \TingRelation $relation
+ *   The relation to handle.
+ *
+ * @return string
+ *   Target attribute for link, "_blank" or "_self".
+ *
+ * @see http://platform.dandigbib.org/issues/378
+ */
+function _opensearch_search_relation_get_target(\TingRelation $relation) {
```

Comment:
Rather than having these provider helpers (target, title, abstract, online url), how about the provider subclasses TingRelation? `ting_relation` can then just call methods on the object, `TingRelation` provides default implementations, and the provider can gather all the specific relation logic in their subclass.

Reply:
I agree if we where talking "raw" access to the data, but in the current situation we're more talking rendering. I'll rename the functions to make that clear.

BUT - I might actually agree that the provider should be able to have it's own subclass here - but in order for it to have that I need to get the the part of the code that does the actual loading of the relations (getRelations on TingEntity and relationsData on TingObject).

I suspect I'll be getting in to that code shortly.